pkg/asset/agent/manifests/agent.go

```diff
@@ -130,7 +130,10 @@ func (m *AgentManifests) validateNMStateLabelSelector() field.ErrorList {
 
 	for _, networkConfig := range m.NMStateConfigs {
 		if !reflect.DeepEqual(m.InfraEnv.Spec.NMStateConfigLabelSelector.MatchLabels, networkConfig.ObjectMeta.Labels) {
```

Comment:
This is actually the wrong test, we should really only be checking that the label selector does actually select those labels. Don't need to fix that now though.